tests/unit/test_tracing.c

```diff
@@ -663,5 +665,99 @@ SENTRY_TEST(drop_unfinished_spans)
+
+    sentry_options_set_traces_sample_rate(options, 1.0);
+    sentry_options_set_max_spans(options, 2);
+    sentry_init(options);
+
+    sentry_transaction_context_t *tx_ctx
+        = sentry_transaction_context_new("wow!", NULL);
+    sentry_transaction_t *tx = sentry_transaction_start(tx_ctx);
+
+    const char *trace_id = sentry_value_as_string(
+        sentry_value_get_by_key(tx->inner, "trace_id"));
+    TEST_CHECK(!sentry__string_eq(trace_id, ""));
+
+    const char *span_id
+        = sentry_value_as_string(sentry_value_get_by_key(tx->inner, "span_id"));
+    TEST_CHECK(!sentry__string_eq(span_id, ""));
+
+    // check transaction
+    tx_ctx = sentry_transaction_context_new("distributed!", NULL);
+    sentry_transaction_iter_headers(tx, forward_headers_to, (void *)tx_ctx);
+    sentry_transaction_t *dist_tx = sentry_transaction_start(tx_ctx);
+
+    const char *dist_trace_id = sentry_value_as_string(
+        sentry_value_get_by_key(dist_tx->inner, "trace_id"));
+    TEST_CHECK_STRING_EQUAL(dist_trace_id, trace_id);
+
+    const char *parent_span_id = sentry_value_as_string(
+        sentry_value_get_by_key(dist_tx->inner, "parent_span_id"));
+    TEST_CHECK_STRING_EQUAL(parent_span_id, span_id);
+
+    sentry__transaction_decref(dist_tx);
+
+    // check span
+    sentry_span_t *child = sentry_transaction_start_child(tx, "honk", "goose");
+
+    span_id = sentry_value_as_string(
+        sentry_value_get_by_key(child->inner, "span_id"));
+    TEST_CHECK(!sentry__string_eq(span_id, ""));
+
+    tx_ctx = sentry_transaction_context_new("distributed!", NULL);
+    sentry_span_iter_headers(child, forward_headers_to, (void *)tx_ctx);
+    dist_tx = sentry_transaction_start(tx_ctx);
+
+    dist_trace_id = sentry_value_as_string(
+        sentry_value_get_by_key(dist_tx->inner, "trace_id"));
+    TEST_CHECK_STRING_EQUAL(dist_trace_id, trace_id);
+
+    parent_span_id = sentry_value_as_string(
+        sentry_value_get_by_key(dist_tx->inner, "parent_span_id"));
+    TEST_CHECK_STRING_EQUAL(parent_span_id, span_id);
+
+    TEST_CHECK(sentry_value_is_true(
+        sentry_value_get_by_key(dist_tx->inner, "sampled")));
+
+    sentry__transaction_decref(dist_tx);
+    sentry__span_free(child);
+    sentry__transaction_decref(tx);
+
+    // check sampled flag
+    tx_ctx = sentry_transaction_context_new("wow!", NULL);
+    sentry_transaction_context_set_sampled(tx_ctx, 0);
+    tx = sentry_transaction_start(tx_ctx);
+
+    tx_ctx = sentry_transaction_context_new("distributed!", NULL);
+    sentry_transaction_iter_headers(tx, forward_headers_to, (void *)tx_ctx);
+    dist_tx = sentry_transaction_start(tx_ctx);
+
+    TEST_CHECK(!sentry_value_is_true(
+        sentry_value_get_by_key(dist_tx->inner, "sampled")));
+
+    sentry__transaction_decref(dist_tx);
+
+    // TODO: Check the sampled flag on a child span as well, but I think we
+    // don't create one if the transaction is not sampled? Well, here is the
+    // reason why we should!
```

Comment:
i'm a little confused as to how this would make for a good use case to carry forward the sampled field into a child span. if it's purely for testing purposes, i can sort of see the argument. it seems like we could just expose unit test-exclusive functionality to check this, though.

if i were to consider the cases outside of the unit test, i'll admit that it's still not too clear to me how and where unsampled spans will be used. i do recognize the fact that other SDKs do still construct spans even if they're unsampled, and perhaps that's enough of an argument to do so.

from the perspective of this feature (distributed trace propagation), a need to continue a trace from an unsampled _span_ and not an unsampled transaction is what's required to make constructing unsampled spans useful. however, it's not clear to me what sort of situations would form the basis of that requirement. are users directly grabbing headers from a span? do we have instrumentation in this SDK that requires grabbing headers from spans?

Reply:
https://develop.sentry.dev/sdk/performance/#propagation

> A transaction's sampling decision should be passed to all of its children, including across service boundaries.

Reply:
sure; the sampling decision does get implicitly propagated by not constructing spans if they're not sampled in the native SDK right now. the only scenario where the native SDK doesn't work is if we're continuing an unsampled span across service boundaries. 

we can continue unsampled transactions across service boundaries in the native SDK, and the one place where you would normally look for a transaction to `iter_headers` off of would be in the scope, which currently exclusively stores a transaction.

do we need to continue unsampled _spans_? do you ever need to invoke `iter_headers` on a span and not a transaction, ie is there an actual use case for calling `iter_headers` on a `sentry_span_t`?

Reply:
either way we could probably retroactively add this in if it's needed, ie it really doesn't affect the primary purpose of this PR (unless not having this straight up breaks distributed tracing but we'd fix that in a follow-up PR anyways)

Reply:
> is there an actual use case for calling `iter_headers` on a `sentry_span_t`?

absolutely! You attach the headers (continue the trace) from whereever you currently are.